cmake/projects/Boost/schemes/url_sha1_boost_library.cmake.in

```diff
@@ -188,8 +188,18 @@ else()
   set(copy_mpi_command "")
 endif()
 
+if(@BUILD_SHARED_LIBS@) #if using shared libs overide boost config
```

Comment:
`if(BUILD_SHARED_LIBS)` should be used here. `@BUILD_SHARED_LIBS@` mean that it will be substituted with the value from the parent project (project that trigger the building of Boost).